clang/include/clang/Sema/Sema.h

```diff
@@ -1763,6 +1763,8 @@ class Sema final : public SemaBase {
   /// Add [[gsl::Pointer]] attributes for std:: types.
   void inferGslPointerAttribute(TypedefNameDecl *TD);
 
+  static bool isPointerLikeType(QualType QT);
```

Comment:
nit: please add some comments for this API. I think it is worth clarifying that it is only used for the lifetime analysis (to avoid misuse).

Reply:
Added comments. Also moved to CheckExprLifetime.h.
I guess that is better to avoid wrong usages.

Reply:
Oh, nice! Moving to `checkExprLifetime.h` makes more sense.